drivers/gles3/storage/material_storage.cpp

```diff
@@ -2601,6 +2614,12 @@ void CanvasShaderData::set_code(const String &p_code) {
 	MaterialStorage::get_singleton()->shaders.canvas_shader.version_set_code(version, gen_code.code, gen_code.uniforms, gen_code.stage_globals[ShaderCompiler::STAGE_VERTEX], gen_code.stage_globals[ShaderCompiler::STAGE_FRAGMENT], gen_code.defines, texture_uniform_data);
 	ERR_FAIL_COND(!MaterialStorage::get_singleton()->shaders.canvas_shader.version_is_valid(version));
 
+	vertex_input_mask = RS::ARRAY_FORMAT_VERTEX;
+	vertex_input_mask |= uses_color << RS::ARRAY_COLOR;
+	vertex_input_mask |= uses_uv << RS::ARRAY_TEX_UV;
```

Comment:
Color and UV are always used by the shader, so they should always be present in the mask. Godot uses them internally, even if users don't access them in their shader. So they have to be present in the mask. 

You can remove the `usage_flag_pointers` for them above too.